From 8b42ffb9c68b222961e29a780ec99a5541dcdaf1 Mon Sep 17 00:00:00 2001
From: arter97 <qkrwngud825@gmail.com>
Date: Sat, 16 May 2015 19:05:05 +0900
Subject: [PATCH 1/3] init: force SELinux to be permissive on all builds

 - Less secure, yes.
 - But also, much stable ROM, too.

And realistically, every temasek ROM users use with root access.
Come on, there's no need to enforce SELinux here.

Signed-off-by: arter97 <qkrwngud825@gmail.com>
---
 init/Android.mk |  4 +---
 init/init.c     | 25 +------------------------
 2 files changed, 2 insertions(+), 27 deletions(-)

diff --git a/init/Android.mk b/init/Android.mk
index 0667d87..6571324 100755
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -25,9 +25,7 @@ LOCAL_SRC_FILES += bootchart.c
 LOCAL_CFLAGS    += -DBOOTCHART=1
 endif
 
-ifneq (,$(filter userdebug eng,$(TARGET_BUILD_VARIANT)))
-LOCAL_CFLAGS += -DALLOW_LOCAL_PROP_OVERRIDE=1 -DALLOW_DISABLE_SELINUX=1
-endif
+LOCAL_CFLAGS += -DALLOW_LOCAL_PROP_OVERRIDE=1
 
 # Enable ueventd logging
 #LOCAL_CFLAGS += -DLOG_UEVENTS=1
diff --git a/init/init.c b/init/init.c
index 1b04af2..3081dfb 100644
--- a/init/init.c
+++ b/init/init.c
@@ -937,29 +937,6 @@ static bool selinux_is_disabled(void)
     return false;
 }
 
-static bool selinux_is_enforcing(void)
-{
-#ifdef ALLOW_DISABLE_SELINUX
-    char tmp[PROP_VALUE_MAX];
-
-    if (property_get("ro.boot.selinux", tmp) == 0) {
-        /* Property is not set.  Assume enforcing */
-        return true;
-    }
-
-    if (strcmp(tmp, "permissive") == 0) {
-        /* SELinux is in the kernel, but we've been told to go into permissive mode */
-        return false;
-    }
-
-    if (strcmp(tmp, "enforcing") != 0) {
-        ERROR("SELinux: Unknown value of ro.boot.selinux. Got: \"%s\". Assuming enforcing.\n", tmp);
-    }
-
-#endif
-    return true;
-}
-
 int selinux_reload_policy(void)
 {
     if (selinux_is_disabled()) {
@@ -1023,7 +1000,7 @@ static void selinux_initialize(void)
     }
 
     selinux_init_all_handles();
-    bool is_enforcing = selinux_is_enforcing();
+    bool is_enforcing = false;
     INFO("SELinux: security_setenforce(%d)\n", is_enforcing);
     security_setenforce(is_enforcing);
 }
-- 
2.5.0

